### PR TITLE
Prefer toolchain fileApi target information over parsing of flags when sending intelliSenseMode to CppTools

### PR DESCRIPTION
**Pull request by andreeis**
Most of this comes from @benmcmorran. I added the undefined check in getIntelliSenseMode from cpptools.ts. We are supposed to let CppTools decide on a default when we don't have target information but when we do we should still send it.

This fixes the following issues:
https://github.com/microsoft/vscode-cmake-tools/issues/1896
https://github.com/microsoft/vscode-cmake-tools/issues/1879

EDIT:
Actually, this PR will be merged only with a fix for 1879. We will create a vsix and send to users that created 1879 and 1896 and if the "--target" issue is not fixed, we may end up implementing the reverse of parseCompilerArgs and parseTargetArch (which would read target information from toolchain fileApi and deduce what is the target switch for the given compiler: -m or -arch or --target or ... etc ...).

### CODE REVIEW

**Comment by Colengms** [comment]
Will this _always_ pass the `--target` arg on to cpptools when present?  If not, I think that would be necessary.  As a compiler arg, it influences the include paths and defines returned by the compiler when we query it.

**Comment by andreeis** [comment]
> Will this _always_ pass the `--target` arg on to cpptools when present? If not, I think that would be necessary. As a compiler arg, it influences the include paths and defines returned by the compiler when we query it.

Yes. We will leave CppTools to decide on a default only when CMake Tools doesn't know anything about the intended arch.

File: src/cpptools.ts [hunk 1]
```diff
@@ -113,6 +113,10 @@ function parseTargetArch(target: string): Architecture {
     return 'arm64';
   } else if (target.indexOf('arm') >= 0 || is_arm_32(target)) {
     return 'arm';
+  } else if (target.indexOf('i686') >= 0) {
+    return 'x86';
+  } else if (target.indexOf('amd64') >= 0 || target.indexOf('x86_64')) {
```
Comment (michelleangela) [inline src/cpptools.ts:118]: Should `target.indexOf('x86_64')` have a check `>= 0`?

Comment (andreeis) [reply src/cpptools.ts:118]: Yes definitely. Thanks for catching that.

**Pull request merged**